### PR TITLE
fix is_virtualenv in Windows

### PR DESCRIPTION
**Pull request by guykisel**
I believe this should fix https://github.com/landscapeio/prospector/issues/159 and https://github.com/landscapeio/prospector/issues/163

I also updated the `is_a_venv` test case so that it'll work properly in Windows.

Slightly unrelated, but you might also be interested to know that I'm using prospector in https://github.com/guykisel/inline-plz-bot so thanks for maintaining such a great tool :smile:

### CODE REVIEW

File: prospector/pathutils.py [hunk 1]
```diff
@@ -8,7 +8,11 @@ def is_virtualenv(path):
     else:
         clues = ('bin', 'lib', 'include')
 
-    dircontents = os.listdir(path)
+    try:
+        dircontents = os.listdir(path)
+    except (OSError, TypeError):
```
Comment (guykisel) [inline prospector/pathutils.py]: The `TypeError` here might seem odd, but without it I get this:
```py27 runtests: commands[0] | nosetests tests
....E........................
======================================================================
ERROR: Windows doesn't allow extremely long paths. This unit test has to be
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\Guy\Documents\prospector\tests\finder\test_file_finder.py", line 50, in test_long_path_not_a_venv
    self.assertFalse(is_virtualenv(path))
  File "C:\Users\Guy\Documents\prospector\prospector\pathutils.py", line 12, in is_virtualenv
    dircontents = os.listdir(path)
TypeError: must be (buffer overflow), not str

----------------------------------------------------------------------
Ran 29 tests in 0.141s```

**Comment by carlio** [comment]
@guykisel Thanks for this! The only machine I have running Windows had a motherboard failure a few months ago so never got around to it myself!

This seems to be related: https://bugs.python.org/issue1776160

Does this therefore work on Windows now? It seems like Python itself is not bothered, if so there's not much that can be done except catch the exception like you have done :)

**Comment by guykisel** [comment]
This does work on Windows now as far as I can tell. I'm testing on Windows 10 with Python 2.7.11.

It does still seem like the original issue from https://github.com/landscapeio/prospector/issues/159 might still exist, where prospector is not honoring `ignore-paths` in Windows. My fix at least makes it a little less painful.

**Comment by guykisel** [comment]
You could maybe set up https://www.appveyor.com/ to get some Windows coverage.

**Comment by guykisel** [comment]
Don't merge yet, I actually have to make the path in the test longer - I forgot that my local repo's path is already a few directories deep.

**Comment by guykisel** [comment]
Updated with a much longer path to guarantee that this test always goes over the Windows maximum path length.

**Comment by carlio** [comment]
@guykisel Finally got around to making a release today, thanks!

**Pull request merged**